fsspec/asyn.py

```diff
@@ -751,6 +755,8 @@ async def _expand_path(self, path, recursive=False, maxdepth=None):
                 if p not in out and (recursive is False or (await self._exists(p))):
                     # should only check once, for the root
                     out.add(p)
+            # reduce depth on each recursion level unless None or 0
+            maxdepth = maxdepth if not maxdepth else maxdepth - 1
```

Comment:
Decrementing `maxdepth` has moved from the beginning to the end of this `else` clause otherwise it is decremented before it is first used.